python/test/fuzzing_test.py

```diff
@@ -0,0 +1,169 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python GLOME fuzz main.
+
+This module that implement some easy fuzz testing for pyglome. 
+"""
+
+import hypothesis
+import unittest
+from cryptography.hazmat.primitives.asymmetric import x25519
+
+import pyglome
+
+
+def _glome(private_bytes, public_bytes, msg, tag, counter, min_tag_len):
+    """Calls basic utilities of glome class, accepts only documented exceptions.
+
+    The intention of this function is to be fuzzed over to find cases that throw
+    unexpected exceptions."""
+    try:
+        private_key = x25519.X25519PrivateKey.from_private_bytes(private_bytes)
+        peer_key = x25519.X25519PublicKey.from_public_bytes(public_bytes)
+    except (ValueError, pyglome.ExchangeError):
+        return
+
+    try:
+        tag_manager = pyglome.Glome(peer_key, private_key, min_tag_len)
+    except ValueError:
+        return
+
+    # Call to property method to test side effects
+    tag_manager.user_keys
+    tag_manager.peer_key
+
+    try:
+        tag_manager.tag(msg, counter)
+    except pyglome.TagGenerationError:
+        pass
+
+    try:
+        tag_manager.check(tag, msg, counter)
+    except (pyglome.IncorrectTagError, pyglome.TagCheckError):
+        pass
+
+
+def _autoglome(private_bytes, public_bytes, msg, tag, counter, min_tag_len,
+               skippable_range):
+    """Calls basic utilities of autoglome class, accepts only documented exceptions.
+
+    The intention of this function is to be fuzzed over to find cases that throw
+    unexpected exceptions."""
+    try:
+        private_key = x25519.X25519PrivateKey.from_private_bytes(private_bytes)
+        peer_key = x25519.X25519PublicKey.from_public_bytes(public_bytes)
+    except (ValueError, pyglome.ExchangeError):
+        return
+
+    try:
+        tag_manager = pyglome.AutoGlome(peer_key,
+                                        private_key,
+                                        min_peer_tag_len=min_tag_len,
+                                        skippable_range=skippable_range)
+    except ValueError:
+        return
+
+    # Call to property method to test side effects
+    tag_manager.user_keys
+    tag_manager.peer_key
+    tag_manager.sending_counter
+    tag_manager.receiving_counter
+
+    try:
+        tag_manager.sending_counter = counter
+    except ValueError:
+        pass
+
+    try:
+        tag_manager.receiving_counter = counter
+    except ValueError:
+        pass
+
+    try:
+        tag_manager.tag(msg)
+    except pyglome.TagGenerationError:
+        pass
+
+    try:
+        tag_manager.check(tag, msg)
+    except (pyglome.IncorrectTagError, pyglome.TagCheckError):
+        pass
+
+
+@hypothesis.settings(max_examples=10**7)
+@hypothesis.given(
+    hypothesis.strategies.binary(min_size=32, max_size=32),  #private_bytes
+    hypothesis.strategies.binary(min_size=32, max_size=32),  #public_bytes
```

Comment:
Have you considered generating incorrect key sizes as well? These should also throw only documented exceptions, but they should probably be fuzzed separately so that we still get coverage on the tagging.

Reply:
Test Added!